py3status/modules/window_title.py

```diff
@@ -2,60 +2,66 @@
 """
 Display window title.
 
-Prints the name of focused window at frequent intervals.
+This module prints the properties of a focused window at frequent intervals.
 
 Configuration parameters:
     cache_timeout: refresh interval for this module (default 0.5)
     format: display format for this module (default '{title}')
-    max_width: If width of title is greater, shrink it and add '...'
-        (default 120)
+
+Format placeholders:
+    {class} class name
+    {instance} instance name
+    {title} title name
+
+Examples:
+```
+# show heart if no title
+window_title {
+    format = '{title}|\u2665'
+}
+
+# specify a length
+window_title {
+    format = '[\?max_length=55 {title}]'
+}
 
 @author shadowprince
 @license Eclipse Public License
 
 SAMPLE OUTPUT
-{'full_text': u'business_plan_final3a.doc'}
+{'full_text': u'business_plan_final_3a.doc'}
 """
 
 from json import loads
 
 
-def find_focused(tree):
-    if type(tree) == list:
-        for el in tree:
-            res = find_focused(el)
-            if res:
-                return res
-    elif type(tree) == dict:
-        if tree['focused']:
-            return tree
-        else:
-            return find_focused(tree['nodes'] + tree['floating_nodes'])
-    return ''
-
-
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 0.5
     format = '{title}'
-    max_width = 120
```

Comment:
Just noticed this - it is quite useful functionality for long titles.

I agree that it is far from ideal.  I think I might just go with what we have here anyhow to keep things simple but we'd need a deprecation.  The other option would be`window_properties['title'] = window_properties['title'][:self.max_length}` which is close enough till we get a better fix

Reply:
Just did a commit. This ok?

Reply:
@tobes ?